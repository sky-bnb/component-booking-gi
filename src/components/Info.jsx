import React from 'react';
import styled from 'styled-components';

export const StyledInfo = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%
  height: 60px;
`;

const StyledHomeAndViews = styled.div`
`;

const StyledLightbulb = styled.img`
  height: 50px;
  width: 50px;
`;

const StyledHomeMindsText = styled.div`
  overflow-wrap: break-word;
  font-family: 'Quicksand', sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.28571em;
  color: rgb(72, 72, 72);
  margin: 0px;
`;

const StyledHomeViewedText = styled.div`
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2857142857142858em;
  margin-top: 6px;
  overflow-wrap: break-word;
  color: #484848;
`;

const Info = (props) => {
  const { numViews } = props;

  return (
    <StyledInfo>
      <StyledHomeAndViews>
        <StyledHomeMindsText>This home is on people's minds.</StyledHomeMindsText>
        <StyledHomeViewedText>
          It's been viewed
          {numViews > 500 ? ' 500+ ' : ` ${numViews} `}
          times in the past week.
        </StyledHomeViewedText>
      </StyledHomeAndViews>
      <StyledLightbulb src="http://127.0.0.1:3004/lightbulb.gif" alt="lightbulb animation" />
    </StyledInfo>
  );
};

export default Info;
