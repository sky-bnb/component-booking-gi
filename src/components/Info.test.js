import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Info, { StyledInfo } from './Info.jsx';

describe('Info Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Info />));

  it('renders the Info component', () => {
    expect(wrapper.find(StyledInfo).length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
