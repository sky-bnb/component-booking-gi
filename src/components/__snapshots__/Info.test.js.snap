// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Component matches the snapshot 1`] = `
<Info__StyledInfo>
  <Info__StyledHomeAndViews>
    <Info__StyledHomeMindsText>
      This home is on people's minds.
    </Info__StyledHomeMindsText>
    <Info__StyledHomeViewedText>
      It's been viewed
       undefined 
      times in the past week.
    </Info__StyledHomeViewedText>
  </Info__StyledHomeAndViews>
  <Info__StyledLightbulb
    alt="lightbulb animation"
    src="http://127.0.0.1:3004/lightbulb.gif"
  />
</Info__StyledInfo>
`;
